### PR TITLE
[APPROVED] add GDPR, GPP, and COPPA support

### PR DESCRIPTION
**Pull request by mscottnelson**
Tickets: IDG-803, IDG-830

### CODE REVIEW

File: modules/33acrossAnalyticsAdapter.js [hunk 1]
```diff
@@ -300,6 +300,22 @@ function createReportFromCache(analyticsCache, completedAuctionId) {
...
+  const gdprConsentData = gdprDataHandler.getConsentData();
+  const gdprApplies = Boolean(gdprConsentData?.gdprApplies);
+  report.gdpr = Number(gdprApplies);
+  if (gdprApplies) {
+    report.gdprConsent = gdprConsentData.consentString || '';
+  }
+
+  if (gppDataHandler.getConsentData()) {
+    report.gpp = gppDataHandler.getConsentData().gpp;
+    report.gppSid = gppDataHandler.getConsentData().applicableSections.join(',');
```
Comment (carlosfelix) [inline modules/33acrossAnalyticsAdapter.js:312]: I think we are sending a POST request, isn't it? if that's the case you don't need to convert the applicableSections to a comma separated string, it's fine to leave them as an array.

File: modules/33acrossAnalyticsAdapter.js [hunk 2]
```diff
@@ -300,6 +300,22 @@ function createReportFromCache(analyticsCache, completedAuctionId) {
     report.usPrivacy = uspDataHandler.getConsentData();
   }
 
+  const gdprConsentData = gdprDataHandler.getConsentData();
+  const gdprApplies = Boolean(gdprConsentData?.gdprApplies);
+  report.gdpr = Number(gdprApplies);
+  if (gdprApplies) {
```
Comment (carlosfelix) [inline modules/33acrossAnalyticsAdapter.js:306]: I think in a few slack conversations it has been mentioned that the gdprApplies flag isn't very reliable, and that we should send the GDPR consent string regardless of the value of the flag. So the `If` statement is not needed. Keep the two parameters but don't have the condition.
```
report.gdpr = Number(gdprApplies);
report.gdprConsent = gdprConsentData.consentString || '';
```

There's actually another PR, that hasn't been merged yet, about removing the condition in the 33x UIM

File: test/spec/modules/33acrossAnalyticsAdapter_spec.js [hunk 3]
```diff
@@ -438,10 +439,22 @@ function mapToBids(auctions) {
...
 function getLocalAssert() {
   function isValidAnalyticsReport(report) {
-    assert.containsAllKeys(report, ['analyticsVersion', 'pid', 'src', 'pbjsVersion', 'auctions']);
+    assert.containsAllKeys(report, ['analyticsVersion', 'pid', 'src', 'pbjsVersion', 'auctions', 'gdpr']);
     if ('usPrivacy' in report) {
       assert.match(report.usPrivacy, /[0|1][Y|N|-]{3}/);
     }
+    assert.oneOf(report.gdpr, [0, 1]);
+    if (report.gdpr === 1) {
+      assert.isString(report.gdprConsent);
```
Comment (carlosfelix) [inline test/spec/modules/33acrossAnalyticsAdapter_spec.js:448]: I don't think we are actually hitting these new asserts, not seeing unit tests that pass those values to this function.

Comment (mscottnelson) [reply test/spec/modules/33acrossAnalyticsAdapter_spec.js:496]: fixed

File: modules/33acrossAnalyticsAdapter.js (hunk 2 above)
Comment (mscottnelson) [reply modules/33acrossAnalyticsAdapter.js]: fixed

File: modules/33acrossAnalyticsAdapter.js (hunk 1 above)
Comment (mscottnelson) [reply modules/33acrossAnalyticsAdapter.js]: fixed

**Pull request merged**